### PR TITLE
Document dev journeys

### LINKED ISSUE #293: Make it less confusing to contribute to our notebooks
We've had at least a couple people try to contribute but not do it quite right, and I don't think anyone outside of the core devs has successfully done it

A few things I've noticed:

1. I think `src` and `build` are really confusing names for the 2 folders, `build` sounds like something that would get generated automatically. I'm planning to reorganize so that the notebooks with output (currently in `build`) are just directly under `ergo/notebooks`, and then the notebooks scrubbed of output (currently in `src` will be in `ergo/notebooks/scrubbed`
2. I don't think it's obvious that [this page](https://ergo.ought.org/en/latest/contrib.html#) in the docs is about contributing to the notebooks specifically. So planning to change the title and maybe add a sentence to make that clear
3. We haven't really documented the process for contributing to notebooks anywhere, so planning to add that. Would cover:
  a. make changes either in Jupyter or Colab
  b. run in Colab and save in `ergo/notebooks`
  c. run `make scrub`
  d. you probably want to diff the scrubbed versions rather than the ones with output
  e. once you make a PR you can use ReviewNB on Github to compare the notebooks (helpful for comparing the ones with output)

It's a bit questionable whether this is worth doing (or whether we should just ignore PRs to the notebooks, or clean them up ourselves), but I think I'd appreciate 1. and 2. seems like it might confuse people who are trying to contribute to ergo core

@djthorne and @stuhlmueller , any thoughts on this before I do it?

### PR DESCRIPTION
**Pull request by brachbach**
Change documentation and organization so that a dev who wants to do one of the following can figure out how to do it more easily:

1. Use ergo (e.g. in their own notebook)
2. Contribute to ergo core
3. Contribute to notebooks in the ergo repo

Changes:

1. Fix #293 
1. Update some things that were out of date, e.g. the old getting started section in the docs.
2. reorganize README. Emphasize things people can interact with, deemphasize philosophy and features.
3. Create seaprate CONTRIBUTING.md.
4. Separate info for contributors into info for contributing to ergo core and info for contributing to ergo notebooks.
5. Add new info about contributing, esp. about contributing to notebooks

This could use one more pass from me, which I'll plan to do tomorrow. But given that there are some pretty big changes, I think makes sense for others to start reviewing this now.

### CODE REVIEW

**Comment by brachbach** [comment]
@steveryb would be great to get your thoughts on this

`core_contrib.rst` explains how to generate and review the docs locally, if you want to take that on :)

**Comment by brachbach** [comment]
@stuhlmueller I'll plan to merge this tonight if you haven't responded by then, FYI

**Comment by brachbach** [comment]
@steveryb even though this is now merged, I'd definitely still appreciate your feedback, or your own PRs iterating on this :)

**Comment by steveryb** [comment]
Sorry, didn't end up being online yesterday, will look now! :)

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -0,0 +1,20 @@
+Ergo is an open source project and we love contributions!
+
+There are [many open issues](https://github.com/oughtinc/ergo/projects/1), including plenty that are [good for newcomers](https://github.com/oughtinc/ergo/issues?q=is%3Aissue+is%3Aopen+label%3A%22good+first+issue%22).
```
Comment (steveryb) [inline CONTRIBUTING.md:3]: nice, love this.

File: README.md [hunk 2]
```diff
@@ -34,15 +34,61 @@ samples = ergo.run(deaths_from_infections, num_samples=5000)
...
+
+4. [How long will the average American spend under lockdown?](notebooks/covid-19-average-lockdown.ipynb)
+
+5. [Assorted COVID predictions](notebooks/assorted-predictions.ipynb)
+
+## Contribute
+
+Ergo is an open source project and we love contributions!
+
+See our [instructions for contributers](CONTRIBUTING.md) for more.
```
Comment (steveryb) [inline README.md:90]: I think this would have solved a lot of my confusion early on, and I love how friendly it is!

File: docs/core_contrib.rst [hunk 3]
```diff
@@ -0,0 +1,55 @@
...
+3. Run linting using ``make lint`` (flake8, mypy, black check)
+4. Run tests using ``make test``
+
+    * To run the tests in ``test_metaculus.py``, you'll need the password to
+      our ``oughttest`` account. If you don't have it, you can ask us for it, 
+      or rely on Travis CI to run those tests for you.
+
+5. Generate docs using ``make docs``, load
+   ``docs/build/html/index.html`` and review the generated docs
+6. Or run all of the above using ``make all``
```
Comment (steveryb) [inline docs/core_contrib.rst:40]: nit: maybe it makes sense to have a `make pr` or something, that runs the above steps without having to run all of them? (unless you want folks to basically run targets pre-PR)

File: README.md (hunk 2 above)
Comment (brachbach) [reply README.md:90]: Sorry it didn't exist when you started!

File: docs/core_contrib.rst (hunk 3 above)
Comment (brachbach) [reply docs/core_contrib.rst:40]: that is in fact what `make all` does -- it only runs ~the above steps, and doesn't run all of our `make` commands :p

Made this issue: https://github.com/oughtinc/ergo/issues/337

**Pull request merged**